tests/test_filenames.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+set -eu
```

Comment:
Maybe not exactly useful now, but I guess we don't have a reason to reject catching subshell errors.
```suggestion
set -eEu
```